pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -785,6 +816,9 @@ func (dsc *ReconcileDaemonSet) podsShouldBeOnNode(
 				// Emit an event so that it's discoverable to users.
 				dsc.eventRecorder.Eventf(ds, corev1.EventTypeWarning, FailedDaemonPodReason, msg)
 				podsToDelete = append(podsToDelete, pod.Name)
+			} else if isPodPreDeleting(pod) {
```

Comment:
why we need add PreparingDelete pods into podsToDelete, if there is some logic which make the pods in the podsToDelete in the first place , is it still the case when next reconcile?

Reply:
Yeah, we should put it into podsToDelete even if it has been set to PreparingDelete in previous reconcile.